android/capacitor/src/main/java/com/getcapacitor/plugin/SystemBars.java

```diff
@@ -0,0 +1,200 @@
+package com.getcapacitor.plugin;
+
+import android.content.pm.PackageInfo;
+import android.content.res.Configuration;
+import android.view.View;
+import android.view.Window;
+import androidx.core.graphics.Insets;
+import androidx.core.view.ViewCompat;
+import androidx.core.view.WindowCompat;
+import androidx.core.view.WindowInsetsCompat;
+import androidx.core.view.WindowInsetsControllerCompat;
+import androidx.webkit.WebViewCompat;
+import com.getcapacitor.Plugin;
+import com.getcapacitor.PluginCall;
+import com.getcapacitor.PluginMethod;
+import com.getcapacitor.annotation.CapacitorPlugin;
+import java.util.Locale;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@CapacitorPlugin
+public class SystemBars extends Plugin {
+
+    static final String STYLE_LIGHT = "LIGHT";
+    static final String STYLE_DARK = "DARK";
+    static final String STYLE_DEFAULT = "DEFAULT";
+    static final String INSET_TOP = "TOP";
+    static final String INSET_BOTTOM = "BOTTOM";
+
+    @Override
+    public void load() {
+        super.load();
+        initSystemBars();
+    }
+
+    private boolean hasFixedWebView() {
+        PackageInfo packageInfo = WebViewCompat.getCurrentWebViewPackage(bridge.getContext());
+        Pattern pattern = Pattern.compile("(\\d+)");
+        Matcher matcher = pattern.matcher(packageInfo.versionName);
+
+        if (!matcher.find()) {
+            return false;
+        }
+
+        String majorVersionStr = matcher.group(0);
+        int majorVersion = Integer.parseInt(majorVersionStr);
+
+        return majorVersion >= 140;
+    }
+
+    private void initSystemBars() {
+        boolean enableInsets = getConfig().getBoolean("enableInsets", true);
+        String style = getConfig().getString("style", STYLE_DEFAULT).toUpperCase();
+        boolean hidden = getConfig().getBoolean("hidden", false);
+
+        if (enableInsets) {
```

Comment:
It might be possible to look at the `viewport-fit=cover` meta tag here. I think that would be the cleanest solution, as that is the [official](https://www.w3.org/TR/css-round-display-1/#descdef-viewport-viewport-fit) way of a developer telling the webview/browser "I'll handle the safe area insets"

Reply:
And if we decide not to do that, I think it should be very clearly stated in the docs that a developer should always handle the safe area insets when targeting the Android platform